docs/environment-variables.md

```diff
@@ -82,6 +82,20 @@ the `--migrate` flag).
 
 Default: https://ipfs.io/ipfs/$something (depends on the IPFS version)
 
+## `IPFS_NS_MAP`
+
+Prewarms namesys cache with static records for deteministic tests and debugging.
+Useful for testing things like DNSLink without real DNS lookup.
+
+Example:
+
+```console
+$ IPFS_NS_MAP="dnslink-test1.example.com:/ipfs/bafkreicysg23kiwv34eg2d7qweipxwosdo2py4ldv42nbauguluen5v6am,dnslink-test2.example.com:/ipns/dnslink-test1.example.com" ipfs daemon
+...
+$ ipfs resolve -r /ipns/dnslink-test2.example.com
+/ipfs/bafkreicysg23kiwv34eg2d7qweipxwosdo2py4ldv42nbauguluen5v6am
+```
```

Comment:
FYSA this is what I ended up adding to get deterministic DNSLink resolution for HTTP Gateway tests.
I believe this is good enough for HTTP Gateway tests, and pretty useful as it will work for IPNS as well.

Rationale: I tried a few approaches but everything felt too complex and not portable (required additional software such as dnsmasq and root access) or too brittle and time-consuming (writing a tool for hijacking syscalls responsible for DNS resolv would work as long `CGO_ENABLED=1`, but it feels like even bigger hack than this one).

ps. If this should not be available outside of test environments, we can make it active only if additional `--enable-ns-map` flag  is passed.